compiler/next/lib/uast/DoWhile.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/uast/DoWhile.h"
+
+#include "chpl/uast/Builder.h"
+
+namespace chpl {
+namespace uast {
+
+
+bool DoWhile::contentsMatchInner(const ASTNode* other) const {
+  const DoWhile* lhs = this;
+  const DoWhile* rhs = (const DoWhile*) other;
+  return lhs->loopContentsMatchInner(rhs) &&
```

Comment:
I was just looking at loopContentsMatchInner and it doesn't seem to check loopBodyChildNum or usesDo fields, please fix that in this PR. (And the count of stmts field, if you add it). As we are adding uAST classes we need to carefully check that all the fields are handled in the new class and the parent classes.
 * contentsMatchInner should check all of the fields for equality (except for ASTNode fields)
 * markUniqueStringsInner should mark all UniqueString fields